### PR TITLE
RHEL (all): Create users at install time for image and edge installers

### PR DESCRIPTION
**Pull request by achilleas-k**
This is a followup to PR #2375.

The changes made for RHEL 8.6 and 9.0 are copied to all the other RHEL distro definitions for compatibility.

**More importantly, this enables the Anaconda users module in all RHEL 8.5 image-installer images, which makes the image type usable when built in the service where user customizations are not supported.**

Summary of changes:

- Enabled the user module in Anaconda for the image-installer in all RHEL distros.
- Enabled the user module in Anaconda for the edge-installer when users are defined in the installer blueprint (which is now allowed in all RHEL distros).
- Added RHEL 8.4 `rhel-edge-installer` to test cases.
- Refactoring of older distro definition internals:
    - Cleaned up redundant (duplicated) stage option constructors (the ones relevant to this PR) from inside distro packages.
    - Renamed `tarInstaller` instances to `imageInstaller` to match user-facing name.

See individual commit messages for details.

These changes were discussed with the image definition team and it was agreed that the image definition changes are acceptable:
- For the image-installer, creating users at install time through the kickstart file is functionally equivalent to creating them in the payload and therefore (mostly) transparent to the user.  The addition of the users Anaconda module in the general case enables extra functionality for the user and does not affect existing workflows.
- For the edge-installer, the user creation at install time is an additional feature that does not affect existing workflows.
- The removal of the `name` key in the `groups` stage has no effect on the image info because it doesn't exist in the osbuild stage.

-----

This pull request includes:

- [ ] adequate testing for the new functionality or fixed issue
- [ ] adequate documentation informing people about the change such as
  - [ ] submit a PR for the [guides](https://github.com/osbuild/guides) repository if this PR changed any behavior described there: https://www.osbuild.org/guides/

### CODE REVIEW

**Review by gicmo — APPROVED** [review]
👍

File: internal/osbuild1/groups_stage.go [hunk 1]
```diff
@@ -1,13 +1,15 @@
...
+import "github.com/osbuild/osbuild-composer/internal/blueprint"
+
 type GroupsStageOptions struct {
 	Groups map[string]GroupsStageOptionsGroup `json:"groups"`
 }
 
 func (GroupsStageOptions) isStageOptions() {}
 
 type GroupsStageOptionsGroup struct {
-	Name string `json:"name"`
```
[the remark this answers is not in the document]
Comment (achilleas-k) [reply internal/osbuild1/groups_stage.go:10]: :scream: my deepest and sincerest apologies

**Review by thozza — COMMENTED** [review]
Great work, I love all the deduplication... Way to go... 😎 

There is one weird code block added in rhel-84 `distro_v2.go`, which I consider a potential mistake or alternatively, my lack of understanding. 😄 

Also some 90 image test cases were removed by https://github.com/osbuild/osbuild-composer/pull/2516/commits/c40e4f80fdf42084a1c31b6d74310d7cfd2e1f05 and then added by https://github.com/osbuild/osbuild-composer/pull/2516/commits/95bad8b9b4894f16429de138d55c7287cc44f6c6 although the commit messages do not really explain why, so this may be something to check.

The rest is just nice to have 🙂

File: internal/osbuild1/groups_stage.go (hunk 1 above)
[the remark this answers is not in the document]
Comment (achilleas-k) [reply internal/osbuild1/groups_stage.go:10]: For osbuild2 it's removed now.  For osbuild1 it's still used by Fedora.  I didn't touch the Fedora distro definitions for the reasons mentioned in the commits (big rewrite happening in another PR).

File: tools/test-case-generators/distro-arch-imagetype-map.json [hunk 2]
```diff
@@ -1,304 +1,304 @@
...
+    "ppc64le": [
+      "qcow2",
+      "tar"
+    ],
+    "s390x": [
+      "qcow2",
+      "tar"
+    ]
+  }
 }
```
[the remark this answers is not in the document]
Comment (achilleas-k) [reply tools/test-case-generators/distro-arch-imagetype-map.json:305]: oops.  fixing.

File: internal/distro/rhel84/distro_v2.go [hunk 3]
```diff
@@ -256,14 +256,18 @@ func (t *imageTypeS2) pipelines(customizations *blueprint.Customizations, option
 	pipelines = append(pipelines, *t.buildPipeline(repos, packageSetSpecs["build-packages"]))
 
 	if t.bootISO {
-		var kernelPkg rpmmd.PackageSpec
-		for _, pkg := range packageSetSpecs["installer"] {
+		var kernelPkg *rpmmd.PackageSpec = nil
```
[the remark this answers is not in the document]
Comment (achilleas-k) [reply internal/distro/rhel84/distro_v2.go:259]: Right.  My mistake.

File: internal/distro/rhel84/distro_v2.go [hunk 4]
```diff
@@ -314,16 +329,27 @@ func (t *imageTypeS2) ostreeTreePipeline(repos []rpmmd.RepoConfig, packages []rp
...
 
 	if groups := c.GetGroups(); len(groups) > 0 {
-		p.AddStage(osbuild.NewGroupsStage(t.groupStageOptions(groups)))
+		p.AddStage(osbuild.NewGroupsStage(osbuild.NewGroupsStageOptions(groups)))
+	}
+
+	if userOptions, err := osbuild.NewUsersStageOptions(c.GetUsers(), false); err != nil {
+		return nil, err
+	} else if userOptions != nil {
+		p.AddStage(osbuild.NewUsersStage(userOptions))
```
[the remark this answers is not in the document]
Comment (achilleas-k) [reply internal/distro/rhel84/distro_v2.go:338]: Refactoring error.  Thanks for catching it.

**Review by gicmo — APPROVED** [review]
Now even better ;)

**Comment by gicmo** [comment]
PackIt failures are fixed via #2518

**Pull request merged**